### PR TITLE
chore(webdriver): support for Page.authenticate

### PR DESCRIPTION
**Pull request by Lightning00Blade**

### CODE REVIEW

**Review by OrKoN — APPROVED** [review]

**Review by OrKoN — APPROVED** [review]

File: test/TestExpectations.json [hunk 1]
```diff
@@ -2719,6 +2705,13 @@
     "expectations": ["FAIL"],
     "comment": "TODO: add a comment explaining why this expectation is required (include links to issues)"
   },
+  {
+    "testIdPattern": "[network.spec] network Page.authenticate should allow disable authentication",
+    "platforms": ["darwin", "linux", "win32"],
+    "parameters": ["firefox", "webDriverBiDi"],
+    "expectations": ["FAIL"],
+    "comment": "The implememtation does not expect 2 responseCompleted events"
```
Comment (OrKoN) [inline test/TestExpectations.json]: Could you clarify which implementation?

**Review by OrKoN — APPROVED** [review]
Is there anything left to make a changelog change for interception support?

**Comment by Lightning00Blade** [comment]
Yes the last piece before handling edge cases and errors is `page.setExtraHeaders`.

File: test/TestExpectations.json (hunk 1 above)
Comment (Lightning00Blade) [inline test/TestExpectations.json]:
```suggestion
    "comment": "The Puppeteer implementation does not expect 2 responseCompleted events (that AuthRequired triggered)"
```

Comment (Lightning00Blade) [reply test/TestExpectations.json]: Done, its Puppeteer.
Though the (proposed) Fetch Spec change don't feel easy to implement.
